src/contract-read.ts

```diff
@@ -28,6 +34,14 @@ export async function readContract(
     height = networkInfo.height;
   }
 
+  if (contractId in cache) {
+    if (height in cache[contractId]) {
+      const res = JSON.parse(cache[contractId][height]);
```

Comment:
Two things here, the res would benefit from being typed, as well as wrap the json.parse into a try/catch block, in which case we may need to think how to best return errors. Like undefined state and validity as { validity: { error: "error while parsing json" } }, just thinking out loud here.